### PR TITLE
Remove 'http' from package names (Go HTTP-only part 4)

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

There's no point of having 'http' in our package names given that
Armeria is now HTTP-only. We have some inconsistencies such as that
the HTTP-only `DocService` is under the `server` package while some
other HTTP-only services are under the `server.http` package.

Modifications:

- Remove 'http' from our package names and directory structures
  completely by merging the 'http' packages to their parents
- Miscellaneous:
  - Remove unused files and deprecated classes found during this
    refactoring
  - Make some classes final (which I forgot to do)

Result:

The project is organized better.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/HeaderDateTimeFormat.java [hunk 1]
```diff
@@ -13,21 +13,9 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-/*
```
Comment (imasahiro) [inline core/src/main/java/com/linecorp/armeria/common/HeaderDateTimeFormat.java]: I'm not a lawyer but do you remove this license header intentionally?

File: core/src/main/java/com/linecorp/armeria/common/HttpHeaderNames.java [hunk 2]
```diff
@@ -13,23 +13,10 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
 
-package com.linecorp.armeria.common.http;
+
+
```
Comment (imasahiro) [inline core/src/main/java/com/linecorp/armeria/common/HttpHeaderNames.java]: Remove empty line?

File: core/src/main/java/com/linecorp/armeria/common/HeaderDateTimeFormat.java (hunk 1 above)
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/HeaderDateTimeFormat.java]: No. IDEA did. Thanks for the heads up!

**Comment by trustin** [comment]
Thanks all for reviewing. :-)

**Pull request merged**